doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+    .. note::
+        The Ensemble requires an allocation.
+    Expand same run settings and parameters to Ensemble members.
+    Specify the parameter expansion strategy via `perm_strategy`.
+
+    Begin by initializing a ``RunSettings`` object to expand to
+    all Models:
+
+    .. code-block:: python
+
+        rs = exp.create_run_settings(exe="python", exe_args="output_my_parameter.py")
+
+    Next, define the parameters to expand to all Models and with values to expand
+    via a expansion strategy:
+
+    .. code-block:: python
+
+        params = {
+            "name": ["Ellie", "John"],
+            "parameter": [2, 11]
+        }
+
+    Finally, initialize an ``Ensemble`` by passing in the ``RunSettings``, `params` and `perm_strategy`:
+
+    .. code-block:: python
+
+        ensemble = exp.create_ensemble("ensemble", params=params, run_settings=rs, perm_strategy="all_perm")
+
+    Notice that `perm_strategy="all_perm"` which means all permutations of the `params` key values will
+    be calculated and distributed across Ensemble members. Here there are four permutations. Therefore,
+    the Ensemble will have four ``Model`` members.
+
+Example 2 : Parameter Expansion with ``RunSettings``, ``BatchSettings`` and `params`
+    Submit the Ensemble as a batch job.
+    Expand identical run settings and parameters to Ensemble members.
+    Declare the parameter expansion strategy via `perm_strategy`.
+
+    Begin by initializing and configuring a ``BatchSettings`` object to
+    run the Ensemble instance:
+
+    .. code-block:: python
+
+        batch_args = {
+            "distribution": "block"
+            "exclusive": None
+        }
+        bs = exp.create_batch_settings(nodes=2,
+                               time="10:00:00",
+                               batch_args=batch_args)
+    The above ``BatchSettings`` object will tell SmartSim to run the Ensemble on two
+    nodes with a timeout of 10 hours.
+
+    Next initialize a ``RunSettings`` object to expand to
+    all Models:
+
+    .. code-block:: python
+
+        rs = exp.create_run_settings(exe="python", exe_args="output_my_parameter.py")
+        rs.set_nodes(1)
+
+    Next, define the parameters to include in all Models and with values to expand
+    via a expansion strategy:
+
+    .. code-block:: python
+
+        params = {
+            "name": ["Ellie", "John"],
+            "parameter": [2, 11]
+        }
+
+    Finally, initialize an ``Ensemble`` by passing in the ``RunSettings``, `params` and `perm_strategy`:
+
+    .. code-block:: python
+
+        ensemble = exp.create_ensemble("ensemble", params=params, run_settings=rs, batch_settings=bs, perm_strategy="step")
+
+    Notice that `perm_strategy="step"` which means values of the `params` key will be
+    grouped into intervals and distributed across Ensemble members. Here there are two groups. Therefore,
+    the Ensemble will have two ``Model`` members.
```

Comment:
I had to read the code to be sure what step strategy did. Should we add a table to each strategy showing the expected groups?